docs/source/developer/committer-notes.rst

```diff
@@ -0,0 +1,162 @@
+Committer Notes
+***************
+
+This section maps out development logistics for core team of OpenDP developers.
+
+.. contents:: |toctitle|
+    :local:
+
+Summary
+=======
+
+* Base all code on a long-lived branch called ``main``, where we enforce a linear history.
+* Do development on short-lived feature branches, which are merged to ``main`` with squash commits.
+* Generate new versions via GitHub Releases, using long-lived release branches.
+
+Rationale
+---------
+
+Our process should be as simple as feasible -- but not simpler! We need to balance developer friendliness with the special requirements of privacy-sensitive software.
+
+We want a clean ``main`` branch that is always in a known good state. It should always be usable for development, and all tests should pass. This allows us to create releases easily, using the most up-to-date code.
+
+We need linear code history, without unanticipated changes introduced by merge commits. This is important so that contributions are validated using the exact code that will land on ``main``.
+
+We need separate release branches to allow bug fixes for previous versions. This is important to support library users who can't upgrade to the latest version of OpenDP yet.
+
+All release tasks should be automated and drivable from GitHub. Creating a release should not require setup of a local environment, and should not have any dependencies on non-standard tools. This is important to allow for delegation of tasks and continuity of the project.
+
+
+Task Tracking
+=============
+
+* Use GitHub Issues to track all tasks. This is helpful to know who's working on what.
+* Use the `OpenDP Development GitHub Project <https://github.com/orgs/opendp/projects/1?card_filter_query=label%3A%22opendp+core%22>`_ queue up tasks for work.
+* Manage all changes using GitHub Pull Requests.
+
+Code Hygiene
+============
+
+* Follow the Rust guidelines for coding style. Code should be formatted using the default settings of rustfmt. (Should we enforce this on PRs?)
```

Comment:
It's friendlier to the contributor to just merge it but I do think we should ask them to run their code through rustfmt. For Java, I've used @reviewdog to leave comments on pull requests when the code style is wrong ( https://github.com/IQSS/dataverse/pull/7911 ). Somehow it seems better to hear this from a computer than a person. 😄

Reply:
Great idea! Added #256.